src/AspNet.Security.OAuth.QQ/QQAuthenticationHandler.cs

```diff
@@ -110,10 +114,16 @@ protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] st
                 return OAuthTokenResponse.Failed(new Exception("An error occurred while retrieving an access token."));
             }
 
-            var payload = JObject.FromObject(QueryHelpers.ParseQuery(await response.Content.ReadAsStringAsync())
-                .ToDictionary(pair => pair.Key, k => k.Value.ToString()));
+            var content = QueryHelpers.ParseQuery(await response.Content.ReadAsStringAsync());
 
-            return OAuthTokenResponse.Success(payload);
+            // HACK Work out the best way to do this with System.Text.Json
+            using (var stream = new MemoryStream())
+            {
+                await CopyPayloadAsync(content, stream);
+
+                var payload = JsonDocument.Parse(stream);
```

Comment:
Is this ever disposed of?

Reply:
Yes, the caller will dispose of the OAuthTokenResponse.